packages/ai-core/src/common/variable-service.ts

```diff
@@ -62,13 +143,26 @@ export interface AIVariableService {
     hasVariable(name: string): boolean;
     getVariable(name: string): Readonly<AIVariable> | undefined;
     getVariables(): Readonly<AIVariable>[];
+    getContextVariables(): Readonly<AIContextVariable>[];
     unregisterVariable(name: string): void;
     readonly onDidChangeVariables: Event<void>;
 
     registerResolver(variable: AIVariable, resolver: AIVariableResolver): Disposable;
     unregisterResolver(variable: AIVariable, resolver: AIVariableResolver): void;
     getResolver(name: string, arg: string | undefined, context: AIVariableContext): Promise<AIVariableResolver | undefined>;
 
+    registerArgumentPicker(variable: AIVariable, argPicker: AIVariableArgPicker): Disposable;
+    unregisterArgumentPicker(variable: AIVariable, argPicker: AIVariableArgPicker): void;
+    getArgumentPicker(name: string, context: AIVariableContext): Promise<AIVariableArgPicker | undefined>;
+
+    registerArgumentCompletionProvider(variable: AIVariable, argPicker: AIVariableArgCompletionProvider): Disposable;
+    unregisterArgumentCompletionProvider(variable: AIVariable, argPicker: AIVariableArgCompletionProvider): void;
+    getArgumentCompletionProvider(name: string): Promise<AIVariableArgCompletionProvider | undefined>;
+
+    registerDropHandler(handler: AIVariableDropHandler): Disposable;
+    unregisterDropHandler(handler: AIVariableDropHandler): void;
+    getDropResult(event: DragEvent, context: AIVariableContext): Promise<AIVariableDropResult>;
```

Comment:
@planger With this change, it looks like this file should be moved to the `browser` folder, as its types now require DOM typing, and it couldn't reasonably be instantiated on the backend.

Reply:
Thanks for raising this issue! You are right, this is an oversight.
I think it'd be best if we could avoid moving the variable service entirely to the frontend. How about adding an additional contribution point with a frontend only binding, and have the variable service on the frontend collect the contributions in addition to what the common variable service does?